lib/Doctrine/DBAL/Driver/SQLSrv/SQLSrvStatement.php

```diff
@@ -150,9 +150,13 @@ public function bindParam($column, &$variable, $type = null, $length = null)
      */
     public function closeCursor()
     {
-        if ($this->stmt) {
```

Comment:
@morozov I now figured out the issue with DB2 binding and it is quite nasty -.-. The binding error seems to be related due to `$this->_bindParam` having referenced variable values so the fix that worked for me is to change the portion in `execute()` to:

```php
        if ($params === null) {
            ksort($this->_bindParam);

            $params = array();

            foreach ($this->_bindParam as $column => $value) {
                $params[] = $value;
            }
        }
```

This fixes the binding error but in fact we cannot get `db_bind_param()` to work at all because it is not actually binding immediately at call time. Instead when calling `db2_execute()`, the function looks up variables by the names passed to `db2_bind_param` lazily. And even worse, it expects thos variables to be declared in global scope (WTF). In theory we could create global variables on binding with `global` keyword but that does not seem to work properly either as I am getting `db2_execute(): Binding Error 1` then. So I would suggest changing the code as shown above, because real parameter binding does not work atm anyways.